File: rust/1st_ex/main.rs
use std::io::stdin;

fn func(s: i32, m: i32, n: i32) -> i32 {
    if s * m > (s - s / 7) * n {
        -1
    } else {
        ((s as f64) * (m as f64) / (n as f64)).ceil() as i32
    }
}

fn init() -> i32 {
    let mut input = String::new();
    stdin().read_line(&mut input).expect("Failed to read line");

    match input.trim().parse::<i32>() {
        Ok(num) if num > 0 => return num,
        _ => panic!("Invalid input"),
    }
}

fn main() {
    let s = init();
    let n = init();
    let m = init();

    println!("{}", func(s, m, n));
}

File: java/1st_ex/main.java
import java.util.Scanner;

public class Survaving {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int S = init(scanner);
        int N = init(scanner);
        int M = init(scanner);

        System.out.println(func(S, N, M));
    }

    private static int init(Scanner scanner) {
        int val = scanner.nextInt();
        if (val <= 0) {
            throw new IllegalArgumentException("Invalid input");
        }
        return val;
    }

    private static int func(int S, int N, int M) {
        if (S * M > (S - S / 7) * N) {
            return -1;
        }
        
        int totalDays = S * M / N;
        if ((S * M % N) > 0) {
            totalDays++;
        }
        return totalDays;
    }
}
